File: backend/backend/src/SignalingComponent/Common/Api/DTOs/SuccessMessage.cs
using System.Text.Json.Serialization;
using backend.WebSocketComponent.Common.Api.DTOs;

namespace backend.SignalingComponent.Common.Api.DTOs;

public class SuccessMessage : ITypedMessage
{
    public static string TypeString => "success";

    [JsonIgnore]
    public string InstanceTypeString => TypeString;

    [JsonPropertyName("requestID")]
    public required string RequestId { get; set; }

    [JsonPropertyName("description")]
    public required string Description { get; set; }
}

File: frontend/src/types/common/SuccessMessage.ts
import { ITypedMessage } from "../ITypedMessage";
import { MessageType } from "../MessageType";

export class SuccessMessage implements ITypedMessage {
    public readonly type = MessageType.SUCCESS;
    public msg: {
        requestID: string;
        description: string;
    };

    public constructor(msg: { requestID: string; description: string }) {
        this.msg = msg;
    }
}
